packit/config/aliases.py

```diff
@@ -175,6 +177,153 @@ def get_branches(
     return branches
 
 
+def get_all_ff_branches(
+    dist_git_branches: Union[list, dict, set, None],
+    default: str = DEFAULT_VERSION,
+    default_dg_branch: str = "main",
+    with_aliases: bool = False,
+) -> set[str]:
+    """
+    Returns a list of target branches that can be fast forwarded merging
+    some other branch according to the given dist_git_branches config key.
+
+    The branches should be specified as in the following example
+        {"rawhide": {"open_pull_request_for": ["f40", "f39"]},
+         "epel9": {}
+        }
+
+    dist_git_branches: dist_git_branches config key; can be a list or dict
+    default: the same as get_branches default parameter
+    default_dg_branch: the same as get_branches default_dg_branch parameter
+    with_aliases: the same as get_branches with_aliases parameter
+    """
+    if not isinstance(dist_git_branches, dict):
+        return set()
+
+    ff_branches = set()
+    for open_pull_request_for in dist_git_branches.values():
+        if OPEN_PULL_REQUEST_FOR_KEY in open_pull_request_for:
+            ff_branches.update(
+                get_branches(
+                    *open_pull_request_for[OPEN_PULL_REQUEST_FOR_KEY],
+                    default=default,
+                    default_dg_branch=default_dg_branch,
+                    with_aliases=with_aliases,
+                ),
+            )
+
+    return ff_branches
+
+
+def get_ff_branches_from(
```

Comment:
I don't have better options but am not sure if `ff_branches` would be well-understood.

Probably depends if this is purely for the diverge branches use case (I would with something really explicit, like `branches_to_open_pr_in`) or should be generic (I would go with something like `target_branches`, `secondary_branches`, `follow_up_branches`?

Reply:
I would pick the suffix `branches_to_open_pr_in` but I think `get_branches_to_open_pr_in` is misleading. We open a PR for all our branches, so I would expect all our branches to be in there. And that is something I find misleading in the config too. I would probably change the key config to `fast_forward_branch_into` or something similar instead.

The others labels I think are just as distant to the config option as it is `ff_branches` and they are more generic so probably I still prefer `fast forwardable branches`; it could be also in the long form, maybe?

But if I have to choose one of the above I would say `target_branches`.

Reply:
+1 for being explicit, including `fast_forward` makes sense to me

Reply:
I went through the code once again, sorry if I confused you, what I meant with my comment was to not shorten the name but include it whole (so that the code is more readable). But to provide more details after re-reviewing:
-  > We open a PR for all our branches, so I would expect all our branches to be in there. And that is something I find misleading in the config too. I would probably change the key config to fast_forward_branch_into or something similar instead.

   I agree with this

- also +1 for keeping consistent the name in the code and the config option
- what about using `dependant_branches`, or mentioning `syncing` (like `branches_to_sync`?)

Reply:
> * also +1 for keeping consistent the name in the code and the config option
> * what about using `dependant_branches`, or mentioning `syncing` (like `branches_to_sync`?)

I asked to @nforro who proposed `open_pull_request_for` and we agreed to rename it into `fast_forward_merge_into`. I renamed the config key and the methods in a similar way.